src/helpers/extensionStatus.js

```diff
@@ -1,55 +1,60 @@
 import { useState, useEffect } from 'react'
-import get from 'lodash/get'
 import { useFlag } from 'cozy-flags'
 
 export const extensionStatuses = {
   checking: 'checking',
   installed: 'installed',
-  notInstalled: 'not-installed'
+  notInstalled: 'not-installed',
+  connected: 'connected'
 }
 
 export const useExtensionStatus = () => {
-  const [installed, setInstalled] = useState(extensionStatuses.notInstalled)
+  const [status, setStatus] = useState(extensionStatuses.notInstalled)
   const extensionCheckDisabled = useFlag('extension-check-disabled')
 
   useEffect(() => {
     if (extensionCheckDisabled) {
       return
     }
 
-    const checkExtensionInstallation = () => {
-      window.postMessage(
-        {
-          message: { source: 'cozy-passwords' }
-        },
-        // This star means that we are sending the message to all
-        // tabs / extensions. This is not ideal, but since we don't know the
-        // cozy extension URI, we can't target it directly
-        '*'
-      )
+    const checkExtensionStatus = () => {
+      const event = document.createEvent('Event')
+      event.initEvent('checkextensionstatus')
+      document.dispatchEvent(event)
     }
 
-    checkExtensionInstallation()
-    const interval = setInterval(checkExtensionInstallation, 1000)
+    checkExtensionStatus()
+    const interval = setInterval(checkExtensionStatus, 1000)
+
+    const handleExtensionInstalled = () => {
+      setStatus(extensionStatuses.installed)
+    }
 
-    const handleInstalled = event => {
-      if (get(event, 'data.message.source') !== 'cozy-extension') {
-        return
-      }
+    const handleExtensionConnected = () => {
+      setStatus(extensionStatuses.connected)
 
-      setInstalled(extensionStatuses.installed)
       cleanup()
     }
 
-    window.addEventListener('message', handleInstalled)
+    document.addEventListener('extensioninstalled', handleExtensionInstalled)
+    document.addEventListener('extensionconnected', handleExtensionConnected)
```

Comment:
It might be safer to add the event listeners before dispatching the first event?

Reply:
I don't think this would cause some problems, but it's actually more logic to to this, yes. See https://github.com/cozy/cozy-passwords/pull/16/commits/2df03508c261dfa9c570e3353358ffc2a77c5d31